### DISCUSSION TITLE
Open new window and Close it using back btn in Android toga/python

### DISCUSSION THREAD

**Discussion by salahshakoori20** [beeware/toga#1410] [Q&A]
Hi
i created a new window from main window in Android application using this code:
        new_window = toga.Window(title='new window')
        self.windows.add(new_window)
        secondary_label = toga.Label(text='you are in a secondary window!')
        box = toga.Box(style=Pack(direction=COLUMN, padding=10))
        box.add(secondary_label)
        new_window.content = box
        new_window.show()
this work  currectly
But how to  close it(using back btn Android) and how to using On_close handler of it (on_close is not implemented?)

**Comment by freakboy3742** [comment]
Android (and mobile platforms) doesn't support the concept of new windows. This is a restriction of the device type - phones don't have the idea of "windows" in the same way that a desktop app does.

**Reply by salahshakoori20** [reply]
Ok, But how to create multiPage, multi activity or optoincontainer for android app? these are very necessary for android.

**Reply by freakboy3742** [reply]
At the moment - you don't. Toga is currently in the early stages of development - especially on mobile platforms. There's a reason the version number is 0.3.0.dev30. It isn't a production-ready, fully documented, fully-featured v1.0 GUI toolkit... yet. We certainly hope to be in the not too distant future, but for now - it's not.

**Reply by salahshakoori20** [reply]
thanks, Good luck

**Reply by t-arn** [reply]
My Android apps support multiple windows. I create a root_box and then in there, the main_box of the page. I then swap out main_box when I want to display a new page

**Reply by t-arn** [reply]
see my "Python Beeware Playground" app in Google Play

**Reply by Razer-0** [reply]
@t-arn I like the approach you used, is this still done the same currently or is it done another way

**Reply by freakboy3742** [reply]
There has been no change, because mobile phones don't support multiple windows with close buttons.

**Reply by Razer-0** [reply]
@freakboy3742  I'm currently trying @t-arn's answer by creating different toga.Boxes and calling them with self.main_window.content = self.box_new

**Reply by t-arn** [reply]
@Razer-0 The basic approach is still the same. I'm using my tatogalib for all my Beeware apps: https://www.tanapro.ch/products/taTogaLib/docs/html/index.html
